### PR TITLE
Make deltree throw errors instead of returning a boolean

### PR DESCRIPTION
**Pull request by mhamilton723**

### CODE REVIEW

File: src/core/env/src/main/scala/FileUtilities.scala [hunk 1]
```diff
@@ -30,10 +30,13 @@ object FileUtilities {
...
-  def delTree(file: File): Boolean =
-    if (!file.exists) true
-    else { if (file.isDirectory) file.listFiles.forall(delTree)
-           file.delete }
+  def delTree(file: File): Unit =
+    if (!file.exists) {
+      throw new IllegalArgumentException(s"File $file does not exist")
+    } else {
+      if (file.isDirectory) {file.listFiles.foreach(delTree)}
+      Files.delete(file.toPath)
```
Comment (drdarshan) [inline src/core/env/src/main/scala/FileUtilities.scala]: Is this class private to our code? If not, can we make it?

File: src/core/env/src/main/scala/FileUtilities.scala [hunk 2]
```diff
@@ -30,10 +30,12 @@ object FileUtilities {
     }
   }
 
-  def delTree(file: File): Boolean =
-    if (!file.exists) true
-    else { if (file.isDirectory) file.listFiles.forall(delTree)
-           file.delete }
+  private[microsoft] def delTree(file: File): Unit =
+    if (!file.exists) {
```
Comment (RatanRSur) [inline src/core/env/src/main/scala/FileUtilities.scala]: I was just reminded of this too but it's more idiomatic to use require for IllegalArgumentExceptions

Comment (mhamilton723) [inline src/core/env/src/main/scala/FileUtilities.scala]: can you write an example

Comment (RatanRSur) [inline src/core/env/src/main/scala/FileUtilities.scala]:
```
private[microsoft] def delTree(file: File): Unit = {
  require(file.exists, s"File $file does not exist")
  FileUtils.deleteDirectory(file)
}
```

**Comment by elibarzilay** [comment]
I don't get it: if it's a simple wrapper around `deleteDirectory` then why not use that directly?

(I thought that this was the agreed plan instead of fixing it.)

**Comment by elibarzilay** [comment]
Closing -- use `forceDelete` instead of changing `delTree`, see #71.

**Pull request closed without merging**